flow/layers/clip_rect_layer.cc

```diff
@@ -50,13 +50,15 @@ void ClipRectLayer::Paint(PaintContext& context) const {
   SkAutoCanvasRestore save(context.internal_nodes_canvas, true);
   context.internal_nodes_canvas->clipRect(clip_rect_,
                                           clip_behavior_ != Clip::hardEdge);
+  context.view_embedder->transformStack->pushClipRect(clip_rect_);
```

Comment:
view_embedder can be null, protect against it in all call sites.

Reply:
done